src/node/interfaces.cpp

```diff
@@ -856,6 +859,20 @@ class MinerImpl : public Mining
         return TestBlockValidity(state, chainman().GetParams(), chainman().ActiveChainstate(), block, chainman().ActiveChain().Tip(), /*fCheckPOW=*/false, check_merkle_root);
     }
 
+    std::unique_ptr<CBlockTemplate> createNewBlock(const CScript& scriptPubKeyIn, bool use_mempool) override
+    {
+        BlockAssembler::Options options;
+
+        // Block resource limits
+        options.nBlockMaxWeight = gArgs.GetIntArg("-blockmaxweight", options.nBlockMaxWeight);
+        if (const auto blockmintxfee{gArgs.GetArg("-blockmintxfee")}) {
+            if (const auto parsed{ParseMoney(*blockmintxfee)}) options.blockMinFeeRate = CFeeRate{*parsed};
+        }
```

Comment:
In commit "rpc: call CreateNewBlock via miner interface" (f375fb9a0dff7ccead80626f4aee853db7799e6f)

I'm not clear how this implementation lines up with the commit description which says "the pool communicates how many extra bytes it needs for its own outputs (payouts, extra commitments, etc). This needs to be substracted from what the user set as -blockmaxweight."

Is the commit description just saying that this implementation is not complete and it will need to be extended in the future? Would be helpful to clarify.

Reply:
Yes, this is a reference to the future change in sv2. I'll see if I can reword it.